lib/state.js

```diff
@@ -872,32 +872,46 @@ class State {
     assert(result.rowCount === 1, 'inserting spot request had incorrect rowCount');
   }
 
-  async reportEbsUsage({region, volumetype, state, totalcount, totalgb}, client) {
-    assert(typeof region === 'string');
-    assert(typeof volumetype === 'string');
-    assert(typeof state === 'string');
-    assert(typeof totalcount === 'number');
-    assert(typeof totalgb === 'number');
+  async reportEbsUsage(rowData) { //, client) {
+    assert(typeof rowData === 'object');
+    assert(Array.isArray(rowData));
+
+    let transaction = await this.beginTransaction();
 
     // We only want to save the most recent data for each region, volume type, and state
-    // tuple, so should update the totalcount, totalgb, and touched column in the event
-    // of a conflict.
+    // tuple, so first clear the table.
+    await this.clearEbsUsage(transaction);
+
+    let values = rowData.map(row => {
+      let {region, volumetype, state, totalcount, totalgb} = row;
+      assert(typeof region === 'string');
+      assert(typeof volumetype === 'string');
+      assert(typeof state === 'string');
+      assert(typeof totalcount === 'number');
+      assert(typeof totalgb === 'number');
+      return `('${region}', '${volumetype}', '${state}', ${totalcount}, ${totalgb})`;
+    }).join(', ');
     let text = [
-      'INSERT INTO ebsusage (region, volumetype, state, totalcount, totalgb)',
-      'VALUES ($1, $2, $3, $4, $5)',
-      'ON CONFLICT (region, volumetype, state)',
-      'DO UPDATE SET totalcount = EXCLUDED.totalcount, totalgb = EXCLUDED.totalgb, touched = now();',
-    ].join(' ');
-    let values = [region, volumetype, state, totalcount, totalgb];
+      'INSERT INTO ebsusage (region, volumetype, state, totalcount, totalgb) VALUES ',
+      ';',
+    ].join(values);
     let name = 'report-ebs-usage';
 
-    let result;
+    let result = await transaction.query(text);
+    await this.commitTransaction(transaction);
+    assert(result.rowCount === rowData.length, 'inserting EBS usage had incorrect rowCount');
+  }
+
+  async clearEbsUsage(client) {
+    let text = 'TRUNCATE TABLE ebsusage;';
+    
     if (client) {
-      result = await client.query({text, values, name});
+      await client.query(text);
     } else {
-      result = await this._pgpool.query({text, values, name});
+      // We should never arrive in this case, since we clear the database and insert
+      // new values in the same transaction.
+      throw new Error('Client should not be undefined in clearEbsUsage');
```

Comment:
It might be easier to just write `await transaction.query('TRUNCATE TABLE ebsusage');` in `reportEbsUsage`?